Horizon/Horizon/Sources/Features/LearningObjects/Assignment/AssignmentDetails/View/AssignmentConfirmationMessages.swift

```diff
@@ -0,0 +1,78 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2025-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+
+protocol AssignmentConfirmationMessages {
+     func makeSubmissionAlertViewModel(
+        isSegmentControlVisible: Bool,
+        isTextSubmission: Bool,
+        onPerformSubmission: @escaping () -> Void
+    ) -> SubmissionAlertViewModel
+
+    func makeDraftAlertViewModel(onDelete: @escaping () -> Void) -> SubmissionAlertViewModel
+    func makeSuccessAlertViewModel(submission: HSubmission?) -> SubmissionAlertViewModel
+}
```

Comment:
This pattern doesn't really fit any of our existing patterns and I'm not sure about the added benefit.